apps/emqx_ft/src/emqx_ft_api.erl

```diff
@@ -110,3 +111,8 @@ format_file_info(
 
 format_timestamp(Timestamp) ->
     iolist_to_binary(calendar:system_time_to_rfc3339(Timestamp, [{unit, second}])).
+
+format_name(NameBin) when is_binary(NameBin) ->
+    NameBin;
+format_name(Name) when is_list(Name) ->
```

Comment:
@keynslug this is a subtle moment, `name` is declared as `file:name()` which is an UTF string, so for non-ASCII names `iolist_to_binary` fails. We should check the file part,